bundler/lib/dependabot/bundler/update_checker/latest_version_finder/dependency_source.rb

```diff
@@ -0,0 +1,149 @@
+# frozen_string_literal: true
+
+module Dependabot
+  module Bundler
+    class UpdateChecker
+      class LatestVersionFinder
+        class DependencySource
+          require_relative "../shared_bundler_helpers"
+          include SharedBundlerHelpers
+
+          RUBYGEMS = "rubygems"
+          PRIVATE_REGISTRY = "private"
+          GIT = "git"
+          OTHER = "other"
+
+          attr_reader :dependency, :dependency_files, :repo_contents_path,
+                      :credentials
+
+          def initialize(dependency:,
+                         dependency_files:,
+                         repo_contents_path: nil,
+                         credentials:)
+            @dependency          = dependency
+            @dependency_files    = dependency_files
+            @repo_contents_path  = repo_contents_path
+            @credentials         = credentials
+          end
+
+          def versions
```

Comment:
How do you feel about adding a small snippet of yard documentation to the public methods that we're introducing? I've often found myself digging into the source code trying to figure out what our objects are meant to be doing, would be nice if we could gradually start introducing some of this:

```suggestion
          # The latest version details for a set of dependencies
          #
          # @return [Array<Hash<Symbol, String>>, Gem::Version]
          def versions
```

Reply:
Yeah, that's a good call